purchase_cancel_reason/views/purchase_order.xml

```diff
@@ -0,0 +1,69 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="purchase_order_form" model="ir.ui.view">
+        <field name="name">purchase.order.form</field>
+        <field name="model">purchase.order</field>
+        <field name="inherit_id" ref="purchase.purchase_order_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//button[@name='button_cancel']" position="attributes">
+                <attribute name="invisible">True</attribute>
+            </xpath>
+            <xpath expr="//button[@name='button_cancel']" position="before">
+                <button
+                    name="%(action_purchase_order_cancel)d"
+                    states="draft,to approve,sent,purchase"
+                    string="Cancel"
+                    type="action"
+                    groups="base.group_user"
+                />
+            </xpath>
+            <xpath expr="/form/sheet/div/h1" position="after">
+                <h5 class="mb-3" attrs="{'invisible': [('state', '!=', 'cancel')]}">
+                    <label for="cancel_reason_id" string="Cancellation reason:" />
+                    <field
+                        name="cancel_reason_id"
+                        class="oe_inline"
+                        options='{"no_open": True}'
+                    />
```

Comment:
I propose to improve the view as follow:
```suggestion
                     <label for="cancel_reason_id"/>
                     <pan> </pan>
                    <field name="cancel_reason_id" class="oe_inline" options="{&quot;no_open&quot;: True}"/>
```
Before:
![image](https://user-images.githubusercontent.com/544090/221131120-91d80dba-350e-4d57-a5ce-30bef2243fd1.png)

After:
![image](https://user-images.githubusercontent.com/544090/221133539-bd25bb17-3861-471b-9f8b-8e47eb2da975.png)


After